site_tools/image_deployer.py

```diff
@@ -0,0 +1,316 @@
+        self.raw_name = name + ".raw"
+        self.qcow_name = name + ".qcow2"
+        self.scope = scope
+        self.base_container = base_container
+        self.current_path = current_path
+        self.container_path = self.base_container + "/" + self.current_path
+
+    def __str__(self):
+        return f"Image(name={self.name})"
+
+
+def get_openstack_connection(cloud, cloud_name):
+    return openstack.connect(auth_url=cloud['clouds'][cloud_name]["auth"]["auth_url"],
+                             auth_type="v3applicationcredential",
+                             application_credential_id=cloud['clouds'][cloud_name]["auth"]["application_credential_id"],
+                             application_credential_secret=cloud['clouds'][cloud_name]["auth"]["application_credential_secret"],
+                             region_name=cloud['clouds'][cloud_name]["region_name"])
+
+
+def get_current_value(connection, base_container, scope):
+    _, current = connection.get_object(base_container, scope + "/current")
+    return current.strip()
+
+
+def get_available_images(connection, base_container, scope, current):
+    logging.info("Checking available images...")
+    available_images = []
+
+    current_path = scope + "/" + current
+    current_objects = connection.object_store.objects(base_container,
+                                                      prefix=current_path,
+                                                      delimiter="/")
+
+    for obj in islice(current_objects, 1, None):
+        object_name = obj.name.split("/")[-1]
+        logging.debug(f"Checking object: {object_name}")
+        if object_name.endswith(".manifest"):
+            name = object_name.rstrip(".manifest")
+            available_images.append(Image(name, base_container, scope, current_path))
+
+    return available_images
+
+
+def get_site_images(connection):
+    logging.info("Checking site images...")
+    images = [i.name for i in connection.image.images()]
+    return images
+
+
+def should_sync_image(image_name, site_images, current):
+    if image_name in site_images:
+        logging.info(f"Image {image_name} already in site images.")
+        image = image_connection.image.find_image(image_name)
+        image_properties = image.properties
+        logging.debug(f"Image properties: {image_properties}")
+        image_current_value = image_properties.get("current", None)
+        logging.info(f"Image {image_name} current value: {image_current_value}")
+        if image_current_value is not None and image_current_value == current:
+            logging.info(f"Image {image_name} is already current.")
+            return False
+    return True
+
+
+def setup_swift_connection(openstack_conn):
+    auth_token = openstack_conn.auth_token
+    storage_url = openstack_conn.object_store.get_endpoint()
+    swift_conn = SwiftConnection(
+        preauthtoken=auth_token,
+        preauthurl=storage_url,
+        auth_version='3'
+    )
+    return swift_conn
```

Comment:
we don't want to depend on having credentials for the object store, we're assuming the bucket is has public read enabled, whether it's swift/s3/whatever.

Reply:
done (left arg for my own testing, we can remove in final version tho)